src/config/mod.rs

```diff
@@ -3206,6 +3211,46 @@ impl ConfigManager for LogConfigManager {
     }
 }
 
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug, OnlineConfig)]
+#[serde(default)]
+#[serde(rename_all = "kebab-case")]
+pub struct MemoryConfig {
+    // Whether enable the heap profiling which may have a bit performance overhead about 2% for the
+    // default sample rate.
+    #[online_config(skip)]
+    pub enable_heap_profiling: bool,
+
+    // Average interval (log base 2) between allocation samples, as measured in bytes of allocation
+    // activity. Increasing the sampling interval decreases profile fidelity, but also decreases
+    // the computational overhead. The default sample interval is 512 KiB (2^19 B).
+    #[online_config(skip)]
+    pub heap_profiling_sample_rate: usize,
```

Comment:
can we just use this parameter only ? if the rate is 0, then it's disable.

Reply:
Or heap_profiling_sample_rate can be Option<ReadableSize> and by default it's None which means it's disabled.

Reply:
okay

Reply:
Maybe it's not friendly for users. Consider a user disabled the heap profiling once, and later when he wants to re-enable it again, he has to figure out the default profiling sample rate. It could be set wrong causing performance issue.

Reply:
As u mentioned, how about assemble and reunite all relevant configurations on `memory` into `MemoryConfig` for integrity ? Including:
- `memory-usage-limit` to `memory.usage-limit`
- `memory-usage-high-water` to `memory.usage-high-watermark`.

Reply:
There are some other configs related to memory spreading out in many components' configs. I decided to do it in later PRs.

Reply:
> Maybe it's not friendly for users. Consider a user disabled the heap profiling once, and later when he wants to re-enable it again, he has to figure out the default profiling sample rate. It could be set wrong causing performance issue.

I think as long as we document the default value, it would not be too confusing for user to set. 
Reducing the parameter count in general is more important than a little bit convenient. High parameter count number  itself is the complexty.